### PR TITLE
CASMINST-6666 only have csm.storage.smartmon redeploy node-exporter if ceph admin keyring is present

### PR DESCRIPTION
**Pull request by leliasen-hpe**
## Summary and Scope

This change fixes the csm.storage.smartmon ansible. The Ceph command to redeploy node-exporter does not work when the ceph.client.admin.keyring is not present on the node. This change will only run the redeploy if the admin keyring is present.

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [CASMINST-6666](https://jira-pro.it.hpe.com:8443/browse/CASMINST-6666)

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * Beau - CSM 1.5 vShasta2

### Test description:

Ran a CFS session on a storage nodes with and without the admin keyring. Ensured the CFS role ran as expected.

When the Ceph admin keyring was present, I saw the following

```bash
TASK [csm.storage.smartmon : Check if ceph.client.admin.keyring exists on node] ***
ok: [x3000c0s29b0n0]

TASK [csm.storage.smartmon : Redeploy node-exporter] ***************************
changed: [x3000c0s29b0n0] => (item=apply -i /etc/cray/ceph/node-exporter.yml)
changed: [x3000c0s29b0n0] => (item=reconfig node-exporter)
changed: [x3000c0s29b0n0] => (item=redeploy node-exporter)
```
 When the Ceph admin keyring was NOT present, I saw the following
 ```bash
TASK [csm.storage.smartmon : Check if ceph.client.admin.keyring exists on node] ***
ok: [x3000c0s30b0n0]

TASK [csm.storage.smartmon : Redeploy node-exporter] ***************************
skipping: [x3000c0s30b0n0] => (item=apply -i /etc/cray/ceph/node-exporter.yml)
skipping: [x3000c0s30b0n0] => (item=reconfig node-exporter)
skipping: [x3000c0s30b0n0] => (item=redeploy node-exporter)
```

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by jsollom-hpe — COMMENTED** [review]
Please add this change to the CHANGELOG.md file.
Was this blowing off because it was being run on other nodes that didn't have this keyring file present? Should it have been targeting other nodes more specifically?
Otherwise, it looks good.

**Review by rbak-hpe — APPROVED** [review]

**Comment by leliasen-hpe** [comment]
> Please add this change to the CHANGELOG.md file. Was this blowing off because it was being run on other nodes that didn't have this keyring file present? Should it have been targeting other nodes more specifically? Otherwise, it looks good.

Yes, this was breaking because it was running on nodes without the ceph admin keyring. This could be run only a single time on a single node. However, we don't want to hardcode a node for it to run on and it doesn't hurt to run it a couple times. The only nodes with that admin keyring are on ncn-s00[1-3] so it won't be running on most storage nodes.

I will add it to the CONTRIBUTING.md file.

**Review by jsollom-hpe — APPROVED** [review]

**Review by jsl-hpe — APPROVED** [review]

File: CHANGELOG.md
Comment (mharding-hpe) [inline CHANGELOG.md:1]: Should also update the link list at the end of the changelog

**Review by mharding-hpe — CHANGES_REQUESTED** [review]

File: ansible/roles/csm.storage.smartmon/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#
+# MIT License
+#
+# (C) Copyright 2021-2023 Hewlett Packard Enterprise Development LP
```
Comment (mharding-hpe) [inline ansible/roles/csm.storage.smartmon/defaults/main.yml:4]: If this is a new file, only need the current year in the copyright header
```suggestion
# (C) Copyright 2023 Hewlett Packard Enterprise Development LP
```

File: CHANGELOG.md
Comment (mharding-hpe) [reply CHANGELOG.md:1]: I added a commit to do this.

**Review by mharding-hpe — APPROVED** [review]
Does this also need to be backported to CSM 1.4? If so, a backport PR will need to be made to `release/1.15` branch

**Comment by leliasen-hpe** [comment]
> Does this also need to be backported to CSM 1.4? If so, a backport PR will need to be made to `release/1.15` branch

No, this is new in CSM 1.5 so no backport is needed.

**Pull request merged**